### PR TITLE
Ocgis

### PR DESCRIPTION
**Pull request by doutriaux1**
@aashish24 can you please take a look at this branch and add the bits to get VTK to use our proj4 to it.

Thanks.

### CODE REVIEW

**Comment by durack1** [comment]
@doutriaux1 looks like the `bots` have hung again.. Might need a restart across the `bot-grid`

**Comment by doutriaux1** [comment]
@aashish24 @danlipsa @sankhesh @chaosphere2112 this is ready to go in. It will help with proj4 upgrade as well since it builds proj4.

**Comment by aashish24** [comment]
Yay!!

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 What other library is using proj4 besides VTK?

File: CMake/cdat_modules/pyparsing_external.cmake [hunk 1]
```diff
@@ -2,5 +2,6 @@
 # create an external project to install MyProxyClient,
 # and configure and build it
 set(nm PYPARSING)
-
+set(OLD OFF)
```
Comment (aashish24) [inline CMake/cdat_modules/pyparsing_external.cmake]: @doutriaux1 can you add some comment here?

File: CMakeLists.txt [hunk 2]
```diff
@@ -529,12 +535,14 @@ include(freetype_pkg)
...
 else()
+  include(jpeg_pkg)
   include(pixman_pkg)
   include(fontconfig_pkg)
   include(curses_pkg)
-  include(tiff_pkg)
+  #include(tiff_pkg)
   include(netcdfplus_pkg)
 endif()
+#include(geotiff_pkg)
```
Comment (aashish24) [inline CMakeLists.txt]: @doutriaux1 take it out if not required please

**Comment by doutriaux1** [comment]
@danlipsa OCGIS is using proj4 which is why I put it in in the first place (here)

File: CMake/cdat_modules/pyparsing_external.cmake (hunk 1 above)
Comment (doutriaux1) [inline CMake/cdat_modules/pyparsing_external.cmake]: @aashish24 I think that bit here tells pipinstaller if we want it to use the `--old-and-unmanageable` option in setup.py which contrarily to what its name says make parallel build manageable...

**Comment by danlipsa** [comment]
@doutriaux1 Makes sense.

**Comment by aashish24** [comment]
@doutriaux1 I am building this will let you know how it goes.

**Comment by aashish24** [comment]
@doutriaux1 everything LGTM

**Comment by aashish24** [comment]
@doutriaux1 just need to update the branch with current master. If all goes well, I will merge it.

**Comment by doutriaux1** [comment]
thanks @aashish24 but did you build with -DCDAT_BUILD_OCGIS=ON ? otherwise I don't think it makes any difference.

**Comment by aashish24** [comment]
I built with both  -DCDAT_BUILD_OCGIS=ON and  -DCDAT_BUILD_OCGIS=OFF and both worked for me :+1:

**Comment by doutriaux1** [comment]
Great!

**Comment by aashish24** [comment]
@doutriaux1 ENUM stuff is failing.

**Comment by doutriaux1** [comment]
odd... but easy enough to fix, thanks will push a fix soon

**Comment by aashish24** [comment]
@doutriaux1 I guess you didn't get time to push a fix?

**Comment by doutriaux1** [comment]
you're right...

**Comment by aashish24** [comment]
@doutriaux1 do you want me to look into it?

**Comment by doutriaux1** [comment]
If you have time that would be great!

**Comment by doutriaux1** [comment]
Thanks! @aashish24

**Comment by aashish24** [comment]
@doutriaux1 we skip egg installation because....? Can you remind me please.

**Comment by durack1** [comment]
@aashish24 one reason is that when building using more than one thread, the `easy_install.pth` file isn't locked and so installs get garbled as concurrent writes lead to packages not being listed

**Comment by aashish24** [comment]
> @aashish24 one reason is that when building using more than one thread, the easy_install.pth file isn't locked and so installs get garbled as concurrent writes lead to packages not being listed

@durack1 we avoid that by making sure that dependencies are set such that only one pip packages gets installed at any given time.

**Comment by aashish24** [comment]
@doutriaux1 I believe I fixed all the issues and resolved the merge conflicts. Once the dashboard runs, I will merge this branch.

**Comment by aashish24** [comment]
@doutriaux1 please have a look at my changes and if looks good, merge it (not sure if you want run it on your local machine).

**Comment by aashish24** [comment]
@doutriaux1 tests failing cannot be related to this branch and since this is your branch originally, I am going to merge it as it LGTM :+1:

**Pull request merged**